### PR TITLE
Fix #338

### LINKED ISSUE #338: Backfill item is created but no model is assigned to it
Copied from: https://bugs.chromium.org/p/chromium/issues/detail?id=656580#c7
OS: Windows (7,8,10), Mac (10.10.5, 10.11.4), Linux (14.04 LTS)

URL: http://www.azurespeed.com/Azure/Download

Steps:
1. Launch Chrome, navigate to above URL and download 7-10 files.
2. Pause each file while downloading from download shelf and navigate to chrome://downloads (7-10 entries is seen)
3. Scroll down the download page and again go back to previous tab (azurespeed.com).
4. Again download any one file, switch download tab (chrome://downloads), scroll up the page and observe.

Actual: Unwanted overlapping of 'Incognito' download entry with the first entry is seen.(Also close icon of that entry is not clickable)

Expected: No such overlapping of download entries should be seen.

This is a regression issue broken in M-55, will soon update bisect info.

Good build : 55.0.2879.0
Bad build : 55.0.2880.0

### PR DESCRIPTION
**Pull request by blasten**
Fixes #338. cc @danbeam

### CODE REVIEW

**Comment by danbeam** [comment]
> This results in an empty physical item that has no model assigned to it.

yeah, that's basically what I'm seeing

File: iron-list.html [hunk 1]
```diff
@@ -1739,27 +1739,24 @@
     },
 
     _createFocusBackfillItem: function() {
-      var pidx, fidx = this._focusedIndex;
-      if (this._offscreenFocusedItem || fidx < 0) {
+      var fidx = this._focusedIndex;
+      var pidx = this._getPhysicalIndex(fidx);
+
+      if (this._offscreenFocusedItem || pidx == null || fidx < 0) {
```
Comment (danbeam) [inline iron-list.html]: so is this `pidx == null` the real fix here?

Comment (danbeam) [inline iron-list.html]: ok, i think i understand now.  the diff is just bigger than i expected (but i now understand why).

**Comment by danbeam** [comment]
LGTM :+1:

**Comment by frankiefu** [comment]
LGTM

**Pull request merged**